### PR TITLE
#RDKC-11869,RDKC-11985: Resolve build issues with existing opensource

### PR DESCRIPTION
**Pull request by relangovan**
Reason for change: 1)Upgraded parodus components to 2.0.0v
                   2)Added cpeabs repo for RDKC tr181 params
                   3)Created webconfig bin and added service
Test Procedure: Webcfg binary must be running
Risks: Low

Signed-off-by: relangovan <Revathy_Elangovan@comcast.com>

### CODE REVIEW

File: src/webcfg_multipart.c [hunk 1]
```diff
@@ -1071,8 +1076,15 @@ WEBCFG_STATUS processMsgpackSubdoc(char *transaction_id)
  * @param[in] nmemb size of delivered data
  * @param[out] data curl response data saved.
 */
+#ifdef BUILD_CAMERA
+size_t writer_callback_fn(void *buffer, size_t size, size_t nmemb, void *datain)
+#else
 size_t writer_callback_fn(void *buffer, size_t size, size_t nmemb, struct token_data *data)
```
Comment (karuna2git) [inline src/webcfg_multipart.c:1082]: as per the spec the https://curl.se/libcurl/c/CURLOPT_WRITEFUNCTION.html

> #include <curl/curl.h>
> 
> size_t write_callback(char *ptr, size_t size, size_t nmemb, void *userdata);
> CURLcode curl_easy_setopt(CURL *handle, CURLOPT_WRITEFUNCTION, write_callback);

The userData must be void*.

Please remove the macro and use the new function header itself as default for all

File: rdk_env.xml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##########################################################################
+-->
+<environment>
+  <components>
+    <webcfg>
+     <dependencies>sdk,curl,cjson,rbus,libsyswrapper,libexchanger,webconfig-framework</dependencies>
```
Comment (karuna2git) [inline rdk_env.xml:24]: sdk,curl, rbus,libsyswrapper  is good enough

File: rdk_build.sh [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+# path to components sources and target
+export RDK_SOURCE_PATH=${RDK_SOURCE_PATH-`readlink -m .`}
+export RDK_TARGET_PATH=${RDK_TARGET_PATH-$RDK_SOURCE_PATH}
+
+# default component name
+export RDK_COMPONENT_NAME=${RDK_COMPONENT_NAME-`basename $RDK_SOURCE_PATH`}
+export RDK_DIR=$RDK_PROJECT_ROOT_PATH
+export RDK_FSROOT_PATH=$RDK_PROJECT_ROOT_PATH/sdk/fsroot/ramdisk
+
+if [ "$XCAM_MODEL" == "SCHC2" ]; then
```
Comment (karuna2git) [inline rdk_build.sh:47]: Line 47 to 55 is not needed

File: rdk_build.sh [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+    -v | --verbose ) set -x ;;
+    -p | --platform ) CC_PLATFORM="$2" ; shift ;;
+    -- ) shift; break;;
+    * ) break;;
+  esac
+  shift
+done
+ARGS=$@
+# component-specific vars
+if [ "$XCAM_MODEL" == "SCHC2" ]; then
```
Comment (karuna2git) [inline rdk_build.sh:99]: define a generic macro; not specific to Devices. Something like,
```
    export CC=${SOURCETOOLCHAIN}gcc
    export CXX=${SOURCETOOLCHAIN}g++
    export GXX=${SOURCETOOLCHAIN}g++
    export AR=${SOURCETOOLCHAIN}ar
    export LD=${SOURCETOOLCHAIN}ld
    export STRIP=${SOURCETOOLCHAIN}strip
```

File: rdk_build.sh [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+    else
+      export LDFLAGS+="-L${RDK_PROJECT_ROOT_PATH}/opensource/lib -lmsgpackc -lcjson -lcurl -lcimplog -ltrower-base64 -lwdmp-c -lwrp-c -llinenoise -L${RDK_PROJECT_ROOT_PATH}/rdklogger/src/.libs -lrdkloggers -L${RDK_PROJECT_ROOT_PATH}/opensource/src/rtmessage/ -lrtMessage -L${RDK_PROJECT_ROOT_PATH}/libsyswrapper/source/.libs/ -lsecure_wrapper -L${RDK_PROJECT_ROOT_PATH}/rbuscore/build/rbus-core/lib/ -lrbus-core -L${RDK_PROJECT_ROOT_PATH}/rbus/build/src/ -lrbus -L${RDK_PROJECT_ROOT_PATH}/webcfg-cpeabs/build/src/ -lcpeabs -Wl,-rpath-link=${RDK_PROJECT_ROOT_PATH}/opensource/lib"
```
Comment (karuna2git) [inline rdk_build.sh:117]: Remove the LDFLAGS.
The CMakelist.txt of webcfg shd be able to link all the needed library for it.
all you have to do is to define rpath

Ex: Add the below into CMAKE command at cmake; ie Line#123
```-DCMAKE_EXE_LINKER_FLAGS="-Wl,-rpath-link,${RDK_FSROOT_PATH}/usr/lib"```

**Review by karuna2git — CHANGES_REQUESTED** [review]

**Review by karuna2git — CHANGES_REQUESTED** [review]
Please refer my comment in the changeset

**Comment by relangovan** [comment]
Will create a new pull request after incorporating the review comments.

**Pull request closed without merging**